pkg/apis/core/v1alpha1/types_shoot.go

```diff
@@ -0,0 +1,837 @@
+	CABundle *string `json:"caBundle,omitempty"`
+	// The client ID for the OpenID Connect client, must be set if oidc-issuer-url is set.
+	// +optional
+	ClientID *string `json:"clientID,omitempty"`
+	// If provided, the name of a custom OpenID Connect claim for specifying user groups. The claim value is expected to be a string or array of strings. This flag is experimental, please see the authentication documentation for further details.
+	// +optional
+	GroupsClaim *string `json:"groupsClaim,omitempty"`
+	// If provided, all groups will be prefixed with this value to prevent conflicts with other authentication strategies.
+	// +optional
+	GroupsPrefix *string `json:"groupsPrefix,omitempty"`
+	// The URL of the OpenID issuer, only HTTPS scheme will be accepted. If set, it will be used to verify the OIDC JSON Web Token (JWT).
+	// +optional
+	IssuerURL *string `json:"issuerURL,omitempty"`
+	// ATTENTION: Only meaningful for Kubernetes >= 1.11
+	// key=value pairs that describes a required claim in the ID Token. If set, the claim is verified to be present in the ID Token with a matching value.
+	// +optional
+	RequiredClaims map[string]string `json:"requiredClaims,omitempty"`
+	// List of allowed JOSE asymmetric signing algorithms. JWTs with a 'alg' header value not in this list will be rejected. Values are defined by RFC 7518 https://tools.ietf.org/html/rfc7518#section-3.1
+	// +optional
+	SigningAlgs []string `json:"signingAlgs,omitempty"`
+	// The OpenID claim to use as the user name. Note that claims other than the default ('sub') is not guaranteed to be unique and immutable. This flag is experimental, please see the authentication documentation for further details. (default "sub")
+	// +optional
+	UsernameClaim *string `json:"usernameClaim,omitempty"`
+	// If provided, all usernames will be prefixed with this value. If not provided, username claims other than 'email' are prefixed by the issuer URL to avoid clashes. To skip any prefixing, provide the value '-'.
+	// +optional
+	UsernamePrefix *string `json:"usernamePrefix,omitempty"`
+}
+
+// AdmissionPlugin contains information about a specific admission plugin and its corresponding configuration.
+type AdmissionPlugin struct {
+	// Name is the name of the plugin.
+	Name string `json:"name"`
+	// Config is the configuration of the plugin.
+	// +optional
+	Config *ProviderConfig `json:"config,omitempty"`
+}
+
+// KubeControllerManagerConfig contains configuration settings for the kube-controller-manager.
+type KubeControllerManagerConfig struct {
+	KubernetesConfig `json:",inline"`
+	// HorizontalPodAutoscalerConfig contains horizontal pod autoscaler configuration settings for the kube-controller-manager.
+	// +optional
+	HorizontalPodAutoscalerConfig *HorizontalPodAutoscalerConfig `json:"horizontalPodAutoscaler,omitempty"`
+	// NodeCIDRMaskSize defines the mask size for node cidr in cluster (default is 24)
+	// +optional
+	NodeCIDRMaskSize *int `json:"nodeCIDRMaskSize,omitempty"`
```

Comment:
Accorging to https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types.
> All public integer fields MUST use the Go (u)int32 or Go (u)int64 types, not (u)int (which is ambiguous depending on target platform). Internal types may use (u)int.